index.js

```diff
@@ -76,10 +78,66 @@ function Choo (opts) {
 
 Choo.prototype.route = function (route, handler) {
   assert.equal(typeof route, 'string', 'choo.route: route should be type string')
-  assert.equal(typeof handler, 'function', 'choo.handler: route should be type function')
+  assert.equal(typeof handler, 'function', 'choo.route: handler should be type function')
   this.router.on(route, handler)
 }
 
+// Register a route to be loaded asynchronously.
+Choo.prototype.experimentalAsyncRoute = function (route, loader) {
+  assert.equal(typeof route, 'string', 'choo.asyncRoute: asyncRoute should be type string')
+  assert.equal(typeof loader, 'function', 'choo.asyncRoute: loader should be type function')
+
+  var IDLE = 0
+  var LOADING = 1
+  var LOADED = 2
+
+  var loadingState = IDLE
+  var renderRoute = null
+  var view = null
+  var self = this
+
+  this.route(route, function (state, emit) {
+    if (!self._asyncProxy) self._initAsyncProxy()
+    // Begin loading the bundle on the first call
+    if (loadingState === IDLE) {
+      emit('choo:async-route-start', state.route)
+      renderRoute = state.route
+      loadingState = LOADING
+
+      var p = loader(onload)
```

Comment:
we don't need to pass in the `onload` callback here if `p` must return a promise anyway